lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -763,6 +840,55 @@ fixtureWalletWith ctx coins0 = do
         expectEventually' ctx balanceAvailable (sum (balance:coins)) dest
         expectEventuallyL ctx balanceAvailable balanceTotal src
 
+-- | fixtureWallets (src, dest) with many transactions
+-- additionally for dest wallet the list of txs [(txTime, txAmt)] is returned
+fixtureWalletManyTxs
+  :: forall t. (EncodeAddress t, DecodeAddress t)
+  => Context t
+  -> [Natural]
+  -- ^ List of tx amounts
+  -> IO (ApiWallet, ApiWallet, [(UTCTime, Natural)])
+  -- ^ (wSrc, wDest, wDestTxList)
+fixtureWalletManyTxs ctx txAmounts = do
```

Comment:
This seems redundant with:

```hs
fixtureWalletWith
    :: forall t. (EncodeAddress t, DecodeAddress t)
    => Context t
    -> [Natural]
    -> IO ApiWallet
```

except for the return types that seems rather awkward to me. I'd rather see two calls to create the transactions and then list them than doing both steps in a single _rather cryptic_ function :grimacing:

Reply:
Actually `fixtureWalletWith` provides wallet with certain UTxO. It construct tx with batches of 10 inputs each so `fixtureWalletWith ctx [1,2]` is a wallet with one tx, but 2 UTxos.

Reply:
Indeed, still doable to do: 

```hs
fixtureWalletWith ctx $ mconcat
    [ replicate 10 1
    , replicate 10 2
    ]
```

:grimacing: ? What do you think :sweat_smile: ?

Reply:
ok, removed`fixtureWalletManyTxs`